storage/service_plans/resolver.go

```diff
@@ -0,0 +1,105 @@
+package service_plans
+
+import (
+	"context"
+	"fmt"
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/storage"
+)
+
+func ResolveSupportedPlatformIDsForPlans(ctx context.Context, plans []*types.ServicePlan, repository storage.Repository) ([]string, error) {
+	var platformTypes map[string]bool
```

Comment:
I still think this `platformTypes` variable can get initialized inline here instead of on line 25: https://github.com/Peripli/service-manager/blob/a4a3388f2a0d91717b704453171296e2cf8798f7/storage/service_plans/resolver.go#L25

Then you can move the temporary `supportedPlatformTypes` variable below inside the actual branch which uses it and be symmetrical to the `if len(platformNames) != 0 ` check above in terms of flow:

```go
//add a criteria for the supported types
if len(platformTypes) != 0 {
	supportedPlatformTypes := make([]string, 0)
	for platform := range platformTypes {
		supportedPlatformTypes = append(supportedPlatformTypes, platform)
	}

	criteria = []query.Criterion{query.ByField(query.InOperator, "type", supportedPlatformTypes...)}
}
```

Reply:
indeed it looks better this way. Thanks!